metaflow/plugins/aws/awslambda/lambda_decorator.py

```diff
@@ -0,0 +1,264 @@
+                package.sha, TransformableObject(package.blob)
+            )
+            cls.package_sha = package.sha
+
+    def package_init(
+        self,
+        flow,  # type: "FlowSpec"
+        step_name,  # type: str
+        environment,  # type: "MetaflowEnvironment"
+        echo,  # type: Callable[..., None]
+    ):
+        # self.step_init(..) should've run by now and populated this and other
+        # attributes. This assert makes mypy happy as it cannot infer that statically.
+        assert self.run_time_limit is not None
+
+        self.lambda_name = lambda_name(
+            self.memory_mb, self.run_time_limit, self.image_uri
+        )
+
+        def echo_with_prefix(msg, err):  # type: (str, bool) -> None
+            echo(
+                "[awslambda decorator] %s"
+                % (
+                    msg,
+                ),
+                err=err,
+            )
+
+        self.lambda_arn = ensure_lambda(
+            name=self.lambda_name,
+            memory_mb=self.memory_mb,
+            timeout_seconds=self.run_time_limit,
+            role_arn=LAMBDA_ROLE_ARN,
+            image_uri=self.image_uri,
+            refresh_image=self.attributes["refresh_image"],
+            echo=echo_with_prefix,
+        )
+
+    def runtime_init(
+        self,
+        flow,  # type: "FlowSpec"
+        graph,  # type: "FlowGraph"
+        package,  # type: "MetaflowPackage"
+        run_id,  # type: str
+    ):
+        if LAMBDA_ROLE_ARN is None:
+            raise LambdaException(
+                "The *@lambda* decorator requires METAFLOW_LAMBDA_ROLE_ARN to be set to a "
+                "lambda role arn"
+            )
+
+        self.flow = flow
+        self.graph = graph
+        self.package = package
+        self.run_id = run_id
+
+    def runtime_task_created(
+        self,
+        datastore,  # type: "MetaflowDataStore"
+        task_id,  # type: str
+        split_index,  # type: Optional[int]
+        input_paths,  # type: List[str]
+        is_cloned,  # type: bool
+    ):
+        if not is_cloned:
+            self._save_package_once(datastore, self.package)
+
+    def task_pre_step(
+        self,
+        step_name,
+        ds,
+        metadata,
+        run_id,
+        task_id,
+        flow,
+        graph,
+        retry_count,
+        max_retries,
+    ):
+        if metadata.TYPE == "local":
+            self.ds_root = ds.root
+        else:
+            self.ds_root = None
+
+        meta = {}
+        meta["aws-lambda-function-name"] = os.environ["AWS_LAMBDA_FUNCTION_NAME"]
+        meta["aws-lambda-log-group-name"] = os.environ["AWS_LAMBDA_LOG_GROUP_NAME"]
+        meta["aws-lambda-log-stream-name"] = os.environ["AWS_LAMBDA_LOG_STREAM_NAME"]
+        meta["aws-lambda-execution-env"] = os.environ["AWS_EXECUTION_ENV"]
+
+        # Unlike the above vars, this is set by our own code in entrypoint.py
+        meta["aws-lambda-request-id"] = os.environ.get("LAMBDA_REQUEST_ID", "")
```

Comment:
Since this is guaranteed to exist - should we just do `os.environ["LAMBDA_REQUEST_ID"]`